tensorboard/plugins/core/core_plugin.py

```diff
@@ -88,14 +91,47 @@ def get_plugin_apps(self):
         '/histograms': self._redirect_to_index,
         '/images': self._redirect_to_index,
     }
-    if self._assets_zip_provider:
-      with self._assets_zip_provider() as fp:
-        with zipfile.ZipFile(fp) as zip_:
-          for path in zip_.namelist():
-            gzipped_asset_bytes = _gzip(zip_.read(path))
-            apps['/' + path] = functools.partial(
+    apps.update(self.get_resource_apps())
+    return apps
+
+  def get_resource_apps(self):
+    apps = {}
+    if not self._assets_zip_provider:
+      return apps
+
+    with self._assets_zip_provider() as fp:
+      with zipfile.ZipFile(fp) as zip_:
+        # First collect the shasums that will later be used. Since hashes are
+        # in the hex, we convert it to base64 for CSP usage.
```

Comment:
Hmm, maybe we should just emit the hashes as base64 in the first place since this is the primary use case?  It's a bit less common than the hex format but would make the logic here in CorePlugin simpler.

Also, if we do it in Vulcanize we can use the nice typesafe Guava APIs and just replace

```java
String hash = Hashing.sha256().hashString(sourceContent, UTF_8).toString();
```

with

```java
// import com.google.common.io.BaseEncoding;
String hash = BaseEncoding.base64().encode(
     Hashing.sha256().hashString(sourceContent, UTF_8).asBytes());
```

Instead of having to pass strings to `codecs`.